### DISCUSSION TITLE
Power up without doing Repl changes program behaviour

### DISCUSSION THREAD

**Discussion by ADenislebel** [micropython/micropython#11211] [RP2040 / Pico] [answered]
I wrote a simple timer program to turn on a light at a given time
and turn is off at a later time. When connected with Thonny the
program runs fine. However when disconnecting and powering
up the pico from a separate 5 volt source the program  runs
as confirmed by flashing the onboard led but fails to initialize 
the light output set in the first few lines of the code. I tried adding
a machine.reset() as the first line of the code but the symptoms
remain the same.
Program listing below unable to use the attach feature
# Timer program
# setup 1 minute timer
import machine
import utime
from machine import Timer,Pin

On_tm = 6				# on time @ 6 am
Off_tm = 19				# off time @ 18 pm
Hour = 6				# start hour
Minute = 0				# start minute
Sec = 0					# seconds
Alt_flg = True			# altrnate action flag
Led_stat = False		# led status chng flag 
Min_flg = False			# minute turn over flag
Led_io = Pin(25,Pin.OUT)	# onboard led
Lit_io = Pin(1,Pin.OUT)		# relay is on GPIO 0

# update timer callback routine every second
def Upd_timer(t):		
    global Minute,Hour,Sec		# re-use same
    global Led_stat,Min_flg		# variables
    
    Sec += 1				# update sec				
    Led_stat = True			# set led flag
    if Sec is 60:			# chk sec turn over
        Min_flg = True		# set min flag
        Sec = 0				# clear seconds
        Minute += 1			# incrmt minute
        if Minute is 60:	# chk min turn over
            Minute = 0 		# clear minute
            Hour += 1		# update hour
            if Hour is 24:	# chk hour turn over
                Hour = 0	# clear hour
    pass                    

# main starts here
Led_io.on()
utime.sleep(2) 
#machine.reset()

Sec_tmr=Timer(period=1000,mode=Timer.PERIODIC, callback=Upd_timer)
print('Start Time:',Hour,Minute)	# show start time
if Hour >= 6 and Hour < 19:			# check on time limits
    Lit_io.on()						# light on 
while True:						# run forever
    if Hour is On_tm:			# is it 6 Am
        Lit_io.on()				# light goes on
    if Hour is Off_tm:			# is it 19 PM
        Lit_io.off()			# light goes off
    if Min_flg is True:			# chk min flag
        print('Time Hr,Min:',Hour,Minute)	# show time
        Min_flg = False			# clr min flag
    if Led_stat is True:		# chk led status
        if Alt_flg is True:		# chk alt flag
            Led_io.on()			# led on
            Alt_flg = False		# chng flag
        else :					# or
            Led_io.off()		# led off
            Alt_flg = True		# chng flag
        Led_stat = False		# clear status
    utime.sleep_ms(500)    		# no rush pause

**Comment by ADenislebel** [comment, marked as answer]
Also tried a machine.soft_reset() as well on entry, did work either

**Reply by ADenislebel** [reply]
Found the problem. I needed to rename my the timer program to main.py as Thonny simply
reloads the program with its current name to the Pico. However the pico boot up process
only looks for a main.py. In my case I had tried before a blink program that remained on 
the pico as main.py. Since my program also flash the led at the same rate it seemed like it was
running when in fact it was the old blink program. When your green you don't see that
shit coming. Hope this might help other newbies with the pico.
Thanks anyway  Denis

**Reply by peterhinch** [reply]
In general it's best not to rename your program. When you're ready to deploy, edit `main.py` to start the program. That way, if you need to edit the program, you can simply comment-out the line in `main.py` and run it at the REPL.
